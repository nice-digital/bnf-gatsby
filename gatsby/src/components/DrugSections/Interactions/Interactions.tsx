import { type FC } from "react";

import { type SlugAndTitle } from "@/utils";

import styles from "../DrugSection.module.scss";
import { InteractionsContent } from "../InteractionsContent/InteractionsContent";
import { BasePot } from "../types";

export type InteractionsProps = BasePot & {
	interactants: SlugAndTitle[];
};

export const Interactions: FC<InteractionsProps> = ({
	interactants,
	potName,
	slug,
}) => {
	return (
		<section aria-labelledby={slug} className={styles.section}>
			<h2 id={slug} dangerouslySetInnerHTML={{ __html: potName }} />
			<InteractionsContent interactants={interactants} />
		</section>
	);
};
