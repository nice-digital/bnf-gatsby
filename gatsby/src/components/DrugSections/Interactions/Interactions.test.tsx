import { render, screen } from "@testing-library/react";

import { Interactions, type InteractionsProps } from "./Interactions";

describe("Interactions", () => {
	const props: InteractionsProps = {
		potName: "Interactions",
		slug: "interactions",
		interactants: [],
	};

	it("should render section with accessible name", () => {
		render(<Interactions {...props} />);

		expect(
			screen.getByRole("region", { name: "Interactions" })
		).toBeInTheDocument();
	});

	it("should render heading 2 with section name", () => {
		render(<Interactions {...props} />);

		expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
			"Interactions"
		);
	});

	it("should match snapshot for single interactant", () => {
		render(
			<Interactions
				{...props}
				interactants={[{ slug: "abacavir", title: "Abacavir" }]}
			/>
		);

		expect(screen.getByRole("region")).toMatchSnapshot();
	});

	it("should match snapshot for multiple interactant", () => {
		render(
			<Interactions
				{...props}
				interactants={[
					{ slug: "abacavir", title: "Abacavir" },
					{ slug: "lamivudine", title: "<em>Lamivudine</em>" },
				]}
			/>
		);

		expect(screen.getByRole("region")).toMatchSnapshot();
	});
});
