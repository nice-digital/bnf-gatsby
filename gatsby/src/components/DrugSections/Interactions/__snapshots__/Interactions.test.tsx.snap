// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interactions should match snapshot for multiple interactant 1`] = `
<section
  aria-labelledby="interactions"
  class="section"
>
  <h2
    id="interactions"
  >
    Interactions
  </h2>
  <p>
    View interactions for:
  </p>
  <ul
    aria-label="Links to each interactant"
  >
    <li>
      <a
        href="/interactions/abacavir/"
      >
        Abacavir
      </a>
    </li>
    <li>
      <a
        href="/interactions/lamivudine/"
      >
        <em>
          Lamivudine
        </em>
      </a>
    </li>
  </ul>
</section>
`;

exports[`Interactions should match snapshot for single interactant 1`] = `
<section
  aria-labelledby="interactions"
  class="section"
>
  <h2
    id="interactions"
  >
    Interactions
  </h2>
  <p>
    <a
      href="/interactions/abacavir/"
    >
      View interactions for
       
      <span>
        Abacavir
      </span>
    </a>
  </p>
</section>
`;
